### PR TITLE
fix: upgrade react-redux to support react 18; bump redux packages

### PR DESCRIPTION
**Pull request by legobeat**
## **Description**

Currently used React version 18 is not officially supported until react-redux v8. Also brings in updates for other redux packages, but holds off on breaking changes for @reduxjs/toolkit v2.

- Upgrade `react-redux` from `4.1.1` to `4.2.1`
  - `4.2.1` was already used transitively so this is effectively a dedupe
- Replace unmaintained `rn-fetch-blob` with `react-native-blob-util`  

## **Related issues**
- #9280

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by legobeat** [comment]
@SocketSecurity ignore npm/react-native-blob-util@0.18.6

network access ok

**Comment by legobeat** [comment]
The import error occurs in a similar way with all of `react-native-blob-util` versions `0.16.4`, `0.18.6`, `0.19.9`.

**Comment by legobeat** [comment]
@SocketSecurity ignore npm/react-native-blob-util@0.19.9

File: patches/redux-persist-filesystem-storage+4.2.0.patch [hunk 1]
```diff
@@ -20,7 +20,7 @@ index 7198fde..d5936df 100644
  
 -  setItem: (key: string, value: string, callback?: (error: ?Error) => void) =>
 +  setItem: (key: string, value: string, isIOS: boolean = false, callback?: (error: ?Error) => void) => {
-     RNFetchBlob.fs
+     ReactNativeBlobUtil.fs
```
Comment (leotm) [inline patches/redux-persist-filesystem-storage+4.2.0.patch:23]: do we need to update line 28 too?
from `RNFetchBlob.ios.excludeFromBackupKey(pathForKey(key));` to 
`ReactNativeBlobUtil.ios.excludeFromBackupKey(pathForKey(key));`

File: ios/Podfile.lock [hunk 2]
```diff
@@ -663,8 +663,8 @@ PODS:
     - Sentry/HybridSDK (= 8.21.0)
   - RNShare (7.3.7):
     - React-Core
-  - RNSVG (12.1.1):
-    - React
+  - RNSVG (15.2.0):
+    - React-Core
```
[the remark this answers is not in the document]
Comment (legobeat) [reply ios/Podfile.lock:667]: Looks like it, yes: #9595.
(Had to use #9545 to be able to perform the update here, so getting that PR in should make these kinds of changes smoother in the future. Probably also want to add consistency-linting in regular CI like we do for `yarn.lock`)

File: patches/redux-persist-filesystem-storage+4.2.0.patch (hunk 1 above)
Comment (legobeat) [reply patches/redux-persist-filesystem-storage+4.2.0.patch:23]: Good catch, fixed!
Should we not expect that to have failed smoke tests, though? :thinking:

File: ios/Podfile.lock (hunk 2 above)
[the remark this answers is not in the document]
Comment (legobeat) [reply ios/Podfile.lock:667]: https://github.com/MetaMask/metamask-mobile/pull/9630

**Review by leotm — DISMISSED** [review]
lgtm!

File: patches/redux-persist-filesystem-storage+4.2.0.patch (hunk 1 above)
Comment (leotm) [reply patches/redux-persist-filesystem-storage+4.2.0.patch:23]: i was thinking this ^ cc @cortisiko could be worth raising this during QA Office Hours

**Comment by legobeat** [comment]
Resolved conflict in `package.json` on `main` rebase - no other changes.

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**